psl/psl-core/src/lib.rs

```diff
@@ -149,6 +138,26 @@ pub fn parse_configuration(
     diagnostics.to_result().map(|_| out)
 }
 
+pub fn parse_configuration_multi_file(
+    files: Vec<(String, SourceFile)>,
+    connectors: ConnectorRegistry<'_>,
+) -> Result<(Files, Configuration), (Files, diagnostics::Diagnostics)> {
+    let mut diagnostics = Diagnostics::default();
+    let mut configuration = Configuration::default();
+
+    let asts = Files::new(files, &mut diagnostics);
+
+    for (_, _, _, ast) in asts.iter() {
+        let out = validate_configuration(ast, &mut diagnostics, connectors);
+        configuration.extend(out);
+    }
```

Comment:
I purposedly avoid creating a `ParserDatabase` instance here, as it performs additional validations that are not retro-compatible with `parse_configuration`, used in `prisma_fmt::get_config`.